src/main/java/core/basesyntax/service/impl/DeepCopyImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitTransaction;
+import core.basesyntax.service.DeepCopy;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DeepCopyImpl implements DeepCopy {
+    @Override
+    public List<FruitTransaction> getDeepCopy(List<FruitTransaction> list) {
+        List<FruitTransaction> copyList = new ArrayList<>(list.size());
+        for (FruitTransaction fruit : list) {
+            copyList.add(new FruitTransaction(fruit.getType(),
```

Comment:
The constructor used here assumes that `FruitTransaction` has a constructor that takes `type`, `fruit`, and `quantity` as parameters. If `FruitTransaction` is meant to be a data class, it's often a good idea to implement a copy constructor or a cloning method within `FruitTransaction` itself to encapsulate the copying logic.